src/main/java/co/cask/gcp/bigquery/BigQuerySink.java

```diff
@@ -189,6 +196,8 @@ public String getOutputFormatClassName() {
                                                            .collect(Collectors.toList()));
       context.record(Collections.singletonList(operation));
     }
+
+    validateSchema(config);
```

Comment:
I'm not sure if the lineage is recorded right when it's called above or if its recorded later. But in either case, it's safer to do this validation earlier to avoid any possible side-effects if the schema is invalid.

Reply:
It will be later. It is collected here but emitted to TMS in destroy of Workflow. Since the invalid schema will fail the pipeline we will not have emitted fll to tms